ccip-sdk/src/execution.ts

```diff
@@ -10,25 +10,28 @@ import {
   type CCIPVersion,
   type ExecutionReport,
   type Lane,
+  type WithLogger,
   ExecutionState,
 } from './types.ts'
 
 /**
  * Pure/sync function to calculate/generate OffRamp.executeManually report for messageIds
  *
+ * @param ctx - Context for logging
  * @param messagesInBatch - Array containing all messages in batch, ordered
  * @param lane - Arguments for leafHasher (lane info)
  * @param messageId - Message ID to prove for manual execution
  * @param merkleRoot - Optional merkleRoot of the CommitReport, for validation
  * @returns ManualExec report arguments
  **/
 export function calculateManualExecProof<V extends CCIPVersion = CCIPVersion>(
+  ctx: WithLogger,
```

Comment:
Can we move this to the last arg and make it optional with `console` by default? Similar to `discoverOffRamp_`, to make it backw compatible

Reply:
Sure. This was done in a time where the `ctx` and `logger` were not optional, which I did to ensure ctx/logger is passed down everywhere needed; after this initial work, I made them optional, but then this function has this optional parameter, so ctx needed to be first; now that it's optional, it can go last